Src/Base/AMReX_Math.H

```diff
@@ -225,6 +226,70 @@ std::uint64_t umulhi (std::uint64_t a, std::uint64_t b)
 }
 #endif
 
+template <typename T>
+AMREX_GPU_HOST_DEVICE AMREX_FORCE_INLINE
+T comp_ellint_1 (T m)
+{
+    // Computing K based on DLMF
+    // https://dlmf.nist.gov/19.8
+    T tol = 1e-12;
+    T a0 = 1.0;
+    T g0 = std::sqrt(1.0 - m);
+    T a, g;
+
+    // Find Arithmetic Geometric mean
+    while(std::abs(a0 - g0) > tol) {
+        a = 0.5*(a0 + g0);
+        g = std::sqrt(a0 * g0);
+
+        a0 = a;
+        g0 = g;
+    }
+
+    return 0.5*pi<T>()/a;
```

Comment:
If m is 0, a is uninitialized.

Reply:
This is a definition issue. In most cases the integrals are defined with m as the argument. In this case with gcc it is k = sqrt(m). I will update it to be consistent and I will additionally add some test figures doing comparisons.

Reply:
@WeiqunZhang in general though do you think this is the right spot for this calculation?

Reply:
Yes, I think so.

Reply:
It seems that we could use `T tol = std::numeric_limits<T>::epsilon();`. If we want to be conservative, we could make it 10x bigger than epsilon. In my testing, it only increases the number of iteration slightly.

Reply:
I think the latest push should resolve these concerns. I have also added a comparison of the AGM algorithm to what is done in Python's ellipm and ellipk routines.

![image](https://github.com/user-attachments/assets/9574cb07-8065-4895-8ef6-463ec16bd2d1)

Reply:
The above figure is the difference between the two cases for the Z-components of the analytical solution to a current loop.